app/javascript/components/widget-transition-controls.jsx

```diff
@@ -0,0 +1,136 @@
+import React, { useEffect, useReducer } from 'react';
+import PropTypes from 'prop-types';
+import { mathMod } from 'ramda';
+
+const prefetchDelay = 1500;
+
+export const shouldRequestWidget = state =>
+  state.msDurationRemaining < Date.now() - state.startTimestamp;
```

Comment:
Is the "<" operator being used to make a comparison between `state.msDurationRemaining` and `Date.now() - state.startTimestamp;` and if so why?

Reply:
This check is used in the interval which handles transitioning between widgets once the timer has run out. `state.msDurationRemaining` is initially the full 20000 millisecond transition time - the comparison here is checking whether the time elapsed since `state.startTimestamp` exceeds  the duration remaining since the last start. The start timestamp and duration are updated if the timer is paused then unpaused.

Writing this comment out made me realize that `state.msDurationRemaining` should be called `state.msRemainingSinceLastStart` for clarity. I'll make those changes now.